lib/logmsg/logmsg.c

```diff
@@ -862,6 +863,27 @@ log_msg_is_tag_by_name(LogMessage *self, const gchar *name)
 /* structured data elements */
 
 static void
+log_msg_sdata_append_key_escaped(GString *result, const gchar *sstr, gssize len)
+{
+  /* The specification does not have any way to escape keys.
+   * The goal is to create syntactically valid structured data fields. */
+  const guchar *ustr = (const guchar *) sstr;
+
+  for (gssize i = 0; i < len; i++)
+    {
+      if (!isascii(ustr[i]) || ustr[i] == '=' || ustr[i] == ' '
```

Comment:
Just a suggestion:
Not sure how much performance penalty is included due to this, but maybe it would worth build a global constant table: ascii->bool which can tell in one step if escape is needed, instead of the 4 comparison above.